File: 0334-increasing-triplet-subsequence/0334-increasing-triplet-subsequence.cpp
class Solution {
public:
    bool increasingTriplet(vector<int>& nums) {
        int l=INT_MAX,r=INT_MAX,n=nums.size();
        for(int i=0;i<n;i++){
            if(nums[i]<=l){
                l=nums[i];
            }
            else if(nums[i]<=r){
                r=nums[i];
            }
            else{
                return true;
            }
        }
        return false;
    }
};

File: 0334-increasing-triplet-subsequence/0334-increasing-triplet-subsequence.py
class Solution:
    def increasingTriplet(self, nums: List[int]) -> bool:
        if len(nums)<3:
            return False
        l, r = inf, inf
        for num in nums:
            if num <= l:
                l = num
            elif num <= r:
                r = num
            else:
                return True     
        return False